### ISSUE TITLE
Unable to execute IMS modules

### ISSUE THREAD

**Issue by redby** [ansible-collections/ibm_zos_ims#82]
##### SUMMARY
I am unable to execute acbgen module. I followed provided example in documentations, but when I submit request I got following response:

FAILED! => {"changed": true, "content": "", "debug": "", "msg": "AttributeError(\"type object 'acbgen' has no attribute 'acbgen'\")", "rc": ""}

I tried also psbgen and dbdgen which is working correctly. It looks that modules referring to modules_utils are not working. Maybe I am missing some kind of configuration. Can anybody help?

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ims_acb_gen

##### ANSIBLE VERSION

```paste below
ansible [core 2.11.2]
  config file = None
  configured module search path = ['/home/redby/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/redby/.local/lib/python3.9/site-packages/ansible
  ansible collection location = /home/redby/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/redby/.local/bin/ansible
  python version = 3.9.1 (default, May 21 2021, 11:40:24) [GCC 8.3.0]
  jinja version = 3.0.1
  libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection      Version
--------------- -------
ibm.ibm_zos_ims 1.0.1
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

5.4.72-microsoft-standard-WSL2 - Controll node
Z/OS 2.4 - Managed node

##### STEPS TO REPRODUCE

```yaml
- name: Playbook for ACBGEN
  
  hosts: ca11
  gather_facts: no
  environment:
      "{{default_environment}}"

  collections:
    - ibm.ibm_zos_core
    - ibm.ibm_zos_ims

  tasks:
    - name: Example of creating ACBs for specific PSBs.
      ims_acb_gen:
        command_input: BUILD
        psb_name:
          - PACCTDBA
        dbd_name:
          - ACCTDBDA
        psb_lib:
          - IMS.PSBLIB
        dbd_lib:
          - IMS.DBDLIB
        acb_lib: IMS.ACBLIB
        reslib:
          - IMSSYS15.SDFSRESL
        steplib:
          - IMSSYS15.SDFSRESL
        build_psb: false
```

##### ACTUAL RESULTS

```paste below
ansible-playbook [core 2.11.2]
  config file = None
  configured module search path = ['/home/redby/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/redby/.local/lib/python3.9/site-packages/ansible
  ansible collection location = /home/redby/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/redby/.local/bin/ansible-playbook
  python version = 3.9.1 (default, May 21 2021, 11:40:24) [GCC 8.3.0]
  jinja version = 3.0.1
  libyaml = True
No config file found; using defaults
host_list declined parsing /mnt/c/users/rm648641/source/IMS_Tools_DevOps/inventory/hosts as it did not pass its verify_file() method
auto declined parsing /mnt/c/users/rm648641/source/IMS_Tools_DevOps/inventory/hosts as it did not pass its verify_file() method
Parsed /mnt/c/users/rm648641/source/IMS_Tools_DevOps/inventory/hosts inventory source with ini plugin
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.

PLAYBOOK: acbgen.yml ***************************************************************************************************
1 plays in acbgen.yml

PLAY [Playbook for ACBGEN] *********************************************************************************************
META: ran handlers

TASK [Example of creating ACBs for specific PSBs.] *********************************************************************
task path: /mnt/c/users/rm648641/source/IMS_Tools_DevOps/acbgen.yml:14
Using module file /home/redby/.ansible/collections/ansible_collections/ibm/ibm_zos_ims/plugins/modules/ims_acb_gen.py
Pipelining is enabled.
<usilca11.lvn.broadcom.net> ESTABLISH SSH CONNECTION FOR USER: mrvra01
<usilca11.lvn.broadcom.net> SSH: EXEC sshpass -d10 ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'User="mrvra01"' -o ConnectTimeout=10 -o ControlPath=/home/redby/.ansible/cp/9da73c7060 usilca11.lvn.broadcom.net '/bin/sh -c '"'"'_BPXK_AUTOCVT=ON _ENCODE_FILE_NEW='"'"'"'"'"'"'"'"'“IBM-1047”'"'"'"'"'"'"'"'"' _ENCODE_FILE_EXISTING='"'"'"'"'"'"'"'"'“IBM-1047”'"'"'"'"'"'"'"'"' /usr/lpp/IBM/cyp/v3r9/pyz/bin/python3 && sleep 0'"'"''
<usilca11.lvn.broadcom.net> (1, b'\n{"changed": true, "content": "", "rc": "", "debug": "", "failed": true, "msg": "AttributeError(\\"type object \'acbgen\' has no attribute \'acbgen\'\\")", "exception": "  File \\"/tmp/ansible_ims_acb_gen_payload_oa27fo_e/ansible_ims_acb_gen_payload.zip/ansible_collections/ibm/ibm_zos_ims/plugins/modules/ims_acb_gen.py\\", line 293, in run_module\\n", "invocation": {"module_args": {"command_input": "BUILD", "psb_name": ["PACCTDBA"], "dbd_name": ["ACCTDBDA"], "psb_lib": ["IDI.MRVRA01.GSE.PSBLIB"], "dbd_lib": ["IDI.MRVRA01.GSE.DBDLIB"], "acb_lib": "SOME.IMS.ACBLIB", "reslib": ["IMSSYS15.SDFSRESL"], "steplib": ["IMSSYS15.SDFSRESL"], "build_psb": false, "compression": null}}}\n', b'')
<usilca11.lvn.broadcom.net> Failed to connect to the host via ssh:
The full traceback is:
  File "/tmp/ansible_ims_acb_gen_payload_oa27fo_e/ansible_ims_acb_gen_payload.zip/ansible_collections/ibm/ibm_zos_ims/plugins/modules/ims_acb_gen.py", line 293, in run_module
fatal: [usilca11.lvn.broadcom.net]: FAILED! => {
    "changed": true,
    "content": "",
    "debug": "",
    "invocation": {
        "module_args": {
            "acb_lib": "IMS.ACBLIB",
            "build_psb": false,
            "command_input": "BUILD",
            "compression": null,
            "dbd_lib": [
                "IMS.DBDLIB"
            ],
            "dbd_name": [
                "ACCTDBDA"
            ],
            "psb_lib": [
                "IMS.PSBLIB"
            ],
            "psb_name": [
                "PACCTDBA"
            ],
            "reslib": [
                "IMSSYS15.SDFSRESL"
            ],
            "steplib": [
                "IMSSYS15.SDFSRESL"
            ]
        }
    },
    "msg": "AttributeError(\"type object 'acbgen' has no attribute 'acbgen'\")",
    "rc": ""
```

**Comment by redby** [comment]
I found it is already fixed in another branch. When can we expect a merge or new version containing this fix?

**Comment by itsBryantP** [comment]
Hi @redby Thanks for reporting the issue. The fix is currently being validated through our internal testing pipeline and we expect to have that completed in the coming week in preparation for a new beta release. In the interim, a workaround would be to downgrade to Ansible version 2.9 for the time being as the issue appears occur with Ansible 2.10+.

**Comment by redby** [comment]
Thank you very much for the confirmation and the proposed workaround.

**Comment by jashnani** [comment]
Hi @redby This issue was fixed in #79 and there is a Beta release available that includes this fix: [release-v1.2.0-beta.1](https://github.com/ansible-collections/ibm_zos_ims/releases/tag/v1.2.0-beta.1). Please let us know if you run into any other issues.
